src/client/src/components/Item.jsx

```diff
@@ -36,70 +38,74 @@ const ListItem = ({ item, itemToDelete, itemToEdit }) => {
         .slice(0, 19)
         .replace("T", " ");
       const timestampNowToReact = timestampNowToDb.slice(10, 19);
-      await updateDoneTimestamp(item.itemId, timestampNowToDb);
-
-      await updateStatusInDb(item.itemId, e.target.checked);
 
       setStatusCompleteTime(`Done at ${timestampNowToReact}`);
       setHideClass("");
       return;
+    } else {
+      updateCheckBoxAction(item.itemId, false);
+
+      setDecorateClass("");
+      setHideClass("hide");
     }
-    await updateStatusInDb(item.itemId, false);
-    setCheckBoxCheck(false);
-    setDecorateClass("");
-    setHideClass("hide");
   };
 
   const editName = async () => {
-    if (editSaveButtonIcon == editIcon) {
+    if (editSaveButtonIcon === editIcon) {
       setReadOnly(false);
       setEditSaveButtonText(saveIcon);
       return;
     }
     setReadOnly(true);
     setEditSaveButtonText(editIcon);
-    let itemNewName = await itemToEdit(item.itemId, newName);
 
-    setNewName(itemNewName);
+    editItemAction(item.itemId, newName);
+  };
+  const deleteItemHandler = () => {
+    deleteItemAction(item.itemId, item.itemName);
   };
 
   useEffect(() => {
-    if (checkBoxCheck) {
+    if (item.status) {
       setDecorateClass("decorate");
-      
-      
-      const timeFromDb = item.doneAt.slice(0, 19).replace("T", " ").slice(10, 19)
-     
-      const timeHours = timeFromDb.slice(0,3)
-      const correctHours = Number(timeHours) + 3
- 
-   const correctTimeToReact = timeFromDb.replace(timeHours, correctHours.toString())
- setStatusCompleteTime(`Done at ${correctTimeToReact}`)
-     
+
+      const timeFromDb = item.doneAt
+        .slice(0, 19)
+        .replace("T", " ")
+        .slice(10, 19);
+
+      const timeHours = timeFromDb.slice(0, 3);
+      const correctHours = Number(timeHours) + 3;
+
+      const correctTimeToReact = timeFromDb.replace(
+        timeHours,
+        correctHours.toString()
+      );
+      setStatusCompleteTime(`Done at ${correctTimeToReact}`);
     }
-  }, []);
+  }, [item]);
 
   return (
     <li className="list-item flex">
-      {" "}
       <div className="check-box">
         <input
           type="checkBox"
-          checked={checkBoxCheck}
+          checked={item.status}
           onChange={newStatusHandler}
+          onClick={() => updateCheckBoxAction(item.itemId, !item.status)}
```

Comment:
this is not needed, you're already calling this method from the newStatusHandler